u_common.go

```diff
@@ -74,49 +74,68 @@ const (
+	CipherSuites       []uint16       // nil => default
+	CompressionMethods []uint8        // nil => no compression
+	Extensions         []TLSExtension // nil => no extensions
+
+	// GreaseStyle: currently only Chrome(GREASE depends on ClientRandom)
+
+	// sessionID may or may not depend on ticket; nil => random
+	GetSessionID func(ticket []byte) [32]byte
+
+	// TLSFingerprintLink string // ?? link to tlsfingerprint.io for informational purposes
+}
+
 var (
 	// HelloGolang will use default "crypto/tls" handshake marshaling codepath, which WILL
 	// overwrite your changes to Hello(Config, Session are fine).
 	// You might want to call BuildHandshakeState() before applying any changes.
 	// UConn.Extensions will be completely ignored.
-	HelloGolang ClientHelloID = ClientHelloID{helloGolang, helloAutoVers}
+	HelloGolang = ClientHelloID{helloGolang, helloAutoVers}
 
-	// HelloCustom will prepare ClientHello with empty uconn.Extensions so you can fill it with TLSExtension's manually
-	HelloCustom ClientHelloID = ClientHelloID{helloCustom, helloAutoVers}
+	// HelloCustom will prepare ClientHello with empty uconn.Extensions so you can fill it with
+	// TLSExtensions manually or use ApplyPreset function
+	HelloCustom = ClientHelloID{helloCustom, helloAutoVers}
 
 	// HelloRandomized* randomly adds/reorders extensions, ciphersuites, etc.
-	HelloRandomized       ClientHelloID = ClientHelloID{helloRandomized, helloAutoVers}
-	HelloRandomizedALPN   ClientHelloID = ClientHelloID{helloRandomized, helloRandomizedALPN}
-	HelloRandomizedNoALPN ClientHelloID = ClientHelloID{helloRandomized, helloRandomizedNoALPN}
+	HelloRandomized       = ClientHelloID{helloRandomized, helloAutoVers}
+	HelloRandomizedALPN   = ClientHelloID{helloRandomized, helloRandomizedALPN}
+	HelloRandomizedNoALPN = ClientHelloID{helloRandomized, helloRandomizedNoALPN}
 
 	// The rest will will parrot given browser.
-	HelloFirefox_Auto ClientHelloID = ClientHelloID{helloFirefox, helloAutoVers}
-	HelloFirefox_55                 = ClientHelloID{helloFirefox, 55}
-	HelloFirefox_56                 = ClientHelloID{helloFirefox, 56}
+	HelloFirefox_Auto = HelloFirefox_56
+	HelloFirefox_55   = ClientHelloID{helloFirefox, 55}
+	HelloFirefox_56   = ClientHelloID{helloFirefox, 56}
 
-	HelloChrome_Auto ClientHelloID = ClientHelloID{helloChrome, helloAutoVers}
-	HelloChrome_58   ClientHelloID = ClientHelloID{helloChrome, 58}
-	HelloChrome_62   ClientHelloID = ClientHelloID{helloChrome, 62}
-
-	HelloAndroid_Auto        ClientHelloID = ClientHelloID{helloAndroid, helloAutoVers}
-	HelloAndroid_6_0_Browser ClientHelloID = ClientHelloID{helloAndroid, 23}
-	HelloAndroid_5_1_Browser ClientHelloID = ClientHelloID{helloAndroid, 22}
+	HelloChrome_Auto = HelloChrome_62
+	HelloChrome_58   = ClientHelloID{helloChrome, 58}
+	HelloChrome_62   = ClientHelloID{helloChrome, 62}
 )
 
+// based on spec's GreaseStyle, GREASE_PLACEHOLDER may be replaced by another GREASE value
+const GREASE_PLACEHOLDER = 0x0a0a
```

Comment:
is this gonna have to be updated every 6 weeks or whatever?

Reply:
why would it have to be updated every 6 weeks or whatever?

Reply:
I see, this is the template pattern that stays stable. makes sense.